src/lib/indexer/index.ts

```diff
@@ -254,4 +277,59 @@ export default class Indexer {
       }
     })
   }
+  getStatus = async (): Promise<Status> => {
+    const indexerTimeout = this.env.INDEXER_TIMEOUT_MS
+    const currentDate = new Date()
+    if (currentDate.getTime() - this.startupTime.getTime() < indexerTimeout) {
+      // if we started less than 30s ago -> PASS
+      return {
+        status: serviceState.UP,
+        detail: {
+          message: 'Service healthy. Starting up.',
+          startupTime: this.startupTime,
+          latestActivityTime: currentDate,
+        },
+      }
+    }
+    if (this.lastProcessedBlockTime === null && this.lastUnprocessedBlockTime === null) {
+      return {
+        status: serviceState.DOWN,
+        detail: {
+          message: 'Last activity was more than 30s ago, no blocks were processed.',
+          startupTime: this.startupTime,
+          latestActivityTime: null,
+        },
+      }
+    }
+    const latestActivityTime = this.lastProcessedBlockTime || this.lastUnprocessedBlockTime
+    if (latestActivityTime === null) throw new Error('sth')
+    if (currentDate.getTime() - latestActivityTime.getTime() < indexerTimeout) {
+      return {
+        status: serviceState.UP,
+        detail: {
+          message: 'Service healthy. Running.',
+          startupTime: this.startupTime,
+          latestActivityTime: latestActivityTime,
+        },
+      }
+    }
+    if (this.lastProcessedBlockTime) {
+      return {
+        status: serviceState.DOWN,
+        detail: {
+          message: `Last activity was more than 30s ago. Last processed block at : ${this.lastProcessedBlockTime}`,
+          startupTime: this.startupTime,
+          latestActivityTime: latestActivityTime,
+        },
+      }
+    }
+    return {
+      status: serviceState.DOWN,
+      detail: {
+        message: `Last activity was more than 30s ago. Last learned of block: ${this.lastUnprocessedBlockTime}`,
+        startupTime: this.startupTime,
+        latestActivityTime: latestActivityTime,
+      },
```

Comment:
as the only difference between these cases is the message I'd be tempted to combine them and then just set message using a ternary on `this.lastProcessedBlockTime`